### PR TITLE
Switch to PCRE2

### PR DESCRIPTION
**Pull request by vpodzime**
PCRE has been deprecated for a long time now.

Merge together:
https://github.com/cfengine/core/pull/5391
https://github.com/cfengine/buildscripts/pull/1343

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a PR! Maybe @craigcomstock can review this?

File: deps-packaging/pcre2/cfbuild-pcre2.spec [hunk 1]
```diff
@@ -0,0 +1,93 @@
+%define pcre2_version 10.42
+
+Summary: CFEngine Build Automation -- pcre2
+Name: cfbuild-pcre2
+Version: %{version}
+Release: 1
+Source0: pcre2-%{pcre2_version}.tar.gz
+License: MIT
+Group: Other
+Url: http://example.com/
```
Comment (craigcomstock) [inline deps-packaging/pcre2/cfbuild-pcre2.spec:10]: fill this out?

**Review by craigcomstock — COMMENTED** [review]
good start. good luck with the exotics! :)

File: deps-packaging/pcre2/cfbuild-pcre2.spec (hunk 1 above)
Comment (vpodzime) [reply deps-packaging/pcre2/cfbuild-pcre2.spec:10]: That's what we use in our `cfbuild` packages, AFAICT.

**Review by olehermanse — APPROVED** [review]

**Pull request merged**